### ISSUE TITLE
Upgrading to wordpress 5.4 breaks new/edit post or page

### ISSUE THREAD

**Issue by matticoli** [docker-library/wordpress#486]
After upgrading to wordpress 5.4, creating new posts/editing old posts stops working
![image](https://user-images.githubusercontent.com/13091791/79626099-84b7b500-80fb-11ea-8d87-c53e28111d43.png)

Attempt Recovery does nothing, Copy Error provides the following text:
```
345/ht<@https://example.com/wp-includes/js/dist/edit-post.min.js?ver=99ca111a562584048ddeb818a39c18f6:6:28438
je@https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:78:476
ph@https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:215:173
lh@https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:126:411
O@https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:121:71
ze@https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:118:14
mg/<@https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:53:49
unstable_runWithPriority@https://example.com/wp-includes/js/dist/vendor/react.min.js?ver=16.9.0:26:340
Ma@https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:52:280
mg@https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:52:498
V@https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:52:431
Sb@https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:213:96
Mg@https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:85:82
t@https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1:19198
Tt/</e<@https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1:19283
o/<@https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1:15212
o@https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1:15184
K/y</<@https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1:13576
p@https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1:6765
a/</</<@https://example.com/wp-includes/js/dist/redux-routine.min.js?ver=7b7d8224ca3d5f24509583833d71c388:1:9535
I/</<@https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1:8752
A/</</<@https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1:9187
dispatch@https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1:8484
i/<@https://example.com/wp-includes/js/dist/redux-routine.min.js?ver=7b7d8224ca3d5f24509583833d71c388:1:9181
e/<@https://example.com/wp-includes/js/dist/redux-routine.min.js?ver=7b7d8224ca3d5f24509583833d71c388:1:5078
e@https://example.com/wp-includes/js/dist/redux-routine.min.js?ver=7b7d8224ca3d5f24509583833d71c388:1:5053
u/<@https://example.com/wp-includes/js/dist/redux-routine.min.js?ver=7b7d8224ca3d5f24509583833d71c388:1:5007
332/r.any@https://example.com/wp-includes/js/dist/redux-routine.min.js?ver=7b7d8224ca3d5f24509583833d71c388:1:5604
e/<@https://example.com/wp-includes/js/dist/redux-routine.min.js?ver=7b7d8224ca3d5f24509583833d71c388:1:5078
e@https://example.com/wp-includes/js/dist/redux-routine.min.js?ver=7b7d8224ca3d5f24509583833d71c388:1:5053
u/<@https://example.com/wp-includes/js/dist/redux-routine.min.js?ver=7b7d8224ca3d5f24509583833d71c388:1:5001
```

In the JS console, I can also see the following errors:
```
JQMIGRATE: Migrate is installed, version 1.4.1 load-scripts.php:8:552
TypeError: "Object(...)(...) is null"
    mo https://example.com/wp-includes/js/dist/edit-post.min.js?ver=99ca111a562584048ddeb818a39c18f6:6
    je https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:78
    ph https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:215
    lh https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:126
    O https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:121
    ze https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:118
    mg https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:53
    unstable_runWithPriority https://example.com/wp-includes/js/dist/vendor/react.min.js?ver=16.9.0:26
    Ma https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:52
    mg https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:52
    V https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:52
    Sb https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:213
    Mg https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:85
    t https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1
    e https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1
    o https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1
    o https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1
    y https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1
    p https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1
    a Redux
    I https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1
    A https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1
    dispatch https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1
    Redux 8
react-dom.min.js:103:487
TypeError: "l is undefined"
    ht https://example.com/wp-includes/js/dist/edit-post.min.js?ver=99ca111a562584048ddeb818a39c18f6:6
    je https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:78
    ph https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:215
    lh https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:126
    O https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:121
    ze https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:118
    mg https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:53
    unstable_runWithPriority https://example.com/wp-includes/js/dist/vendor/react.min.js?ver=16.9.0:26
    Ma https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:52
    mg https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:52
    V https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:52
    Sb https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:213
    Mg https://example.com/wp-includes/js/dist/vendor/react-dom.min.js?ver=16.9.0:85
    t https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1
    e https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1
    o https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1
    o https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1
    y https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1
    p https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1
    a Redux
    I https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1
    A https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1
    dispatch https://example.com/wp-includes/js/dist/data.min.js?ver=e80048e7b309581a40e44b8bd8aaba41:1
```

This is after updating a clean Wordpress install via wp-admin. My docker-compose for generating the container is pretty standard:
```
version: '2'

services:
   db:
     image: mysql:5.7
     volumes:
       - /home/myuser/db_data:/var/lib/mysql
     restart: always
     environment:
       MYSQL_RANDOM_ROOT_PASSWORD: '1'
       MYSQL_DATABASE: wp
       MYSQL_USER: user
       MYSQL_PASSWORD: ******

   wordpress:
     depends_on:
       - db
     image: wordpress:latest
     volumes:
       - /home/myuser/wp_data:/var/www/html
     ports:
       - "8000:80"
     restart: always
     environment:
       WORDPRESS_DB_HOST: db:3306
       WORDPRESS_DB_USER: user
       WORDPRESS_DB_PASSWORD: ******
       WORDPRESS_DB_NAME: wp
```

Everything works fine on 5.3 - updating to 5.4 borked it in every test. Tried using docker volumes instead of mounting a directory on the host to no avail. The worst part is backing up content/settings after the upgrade and re-importing on a clean 5.3 install also breaks it, as the affected files are included in the backup and I am unable to selectively re-import. Currently hoping I have a recent backup on 5.3 ...

Only thread I could find with a similar bug was this one [https://wordpress.org/support/topic/errors-occurred-after-wordpress-5-4-upgrade/](https://wordpress.org/support/topic/errors-occurred-after-wordpress-5-4-upgrade/), but this doesn't seem to be it as it's a clean install of wordpress

**Comment by wglambert** [comment]
What plugins are you using? Can you try disabling them and doing the upgrade?

**Comment by matticoli** [comment]
The issue persists with all plugins disabled, and is only fixed by downgrading back to 5.3.2. The issue also happens on a clean instance (entirely new stack with the docker-compose listed above, new wordpress and db volume)

**Comment by wglambert** [comment]
>The issue also happens on a clean instance

I can't reproduce, might be an issue with your host's environment or the volumes not being removed?

![04_23_20 @7_16_55](https://user-images.githubusercontent.com/26428991/80109520-70781b80-8532-11ea-8af4-d1e583b24517.png)

Maybe try without bind-mounting any volumes
Are you using Docker for Mac/Windows? Anything special like an NFS storage?

**Comment by matticoli** [comment]
Hello - apologies for letting this sit so long. After several months of on-and-off troubleshooting I was finally able to get to the bottom of this. It was indeed an environment issue and not related to Wordpress or Docker. I'm not quite sure what it was but **tl;dr**: using nginx as a reverse proxy does some really strange things with Wordpress. I experienced different (seemingly unrelated) erratic behavior when trying to reverse proxy a different domain to an externally hosted Wordpress installation.

Some additional context, just in case anyone having a similar problem comes across this:

I was able to diagnose the issue by accessing a test container directly via http, bypassing the reverse proxy. The behavior described in the original issue text only occurred when accessing the site via the reverse proxy, and only using Wordpress 5.4 (downgrading to 5.3 resolved the error). While I was unable to pinpoint the specific cause, I believe it has to do with the URL that the site is served through relative to the URL configured in Wordpress. More specifically, I was able to resolve this issue temporarily in Wordpress 5.4 using mod_substitute in apache2 to replace all instances of '127.0.0.1' with the site's external URL. I suspect some change between 5.3 and 5.4 changed the way scripts in the editor view interact with request headers.

Since I need to use nginx for SSL termination, I am currently reverse proxying via nginx (for SSL) -> apache2 (for mod_sub) -> apache2 (serving PHP in the container), which is clunky to say the least. I will likely explore haproxy as a solution for reverse proxy and SSL termination moving forward to cut out the extra steps.
